cpp/src/aztec3/circuits/abis/rollup/merge/merge_rollup_public_inputs.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+#include "aztec3/circuits/abis/append_only_tree_snapshot.hpp"
+#include <aztec3/utils/types/native_types.hpp>
+#include <aztec3/utils/types/circuit_types.hpp>
+#include <aztec3/utils/types/convert.hpp>
+#include <type_traits>
+
+namespace aztec3::circuits::abis {
+
+using aztec3::utils::types::CircuitTypes;
+using aztec3::utils::types::NativeTypes;
+using std::is_same;
+
+template <typename NCT> struct MergeRollupPublicInputs {
+    typedef typename NCT::fr fr;
+    typedef typename NCT::AggregationObject AggregationObject;
+
+    AggregationObject aggregation_object;
+
```

Comment:
Missed:
`rollup_subtree_height`
`rollup_type`

Also the arch diagram calls `aggregation_object` as `end_aggregation_object`